lib/block/account.go

```diff
@@ -52,7 +53,11 @@ func (b *BlockAccount) Save(st *storage.LevelDBBackend) (err error) {
 	if exists {
 		err = st.Set(key, b)
 	} else {
-		// TODO consider to use, [`Transaction`](https://godoc.org/github.com/syndtr/goleveldb/leveldb#DB.OpenTransaction)
+		if b.Balance < common.BaseReserve {
+			err = errors.ErrorInsufficientAmountNewAccount
+			return
+		}
```

Comment:
Could you move this in the consensus rules please ? The `Save` method shouldn't be the one doing this check.

Reply:
I have the same opinion as @Geod24.
This check is redundant and already checked in the `IsWellFormed` method.
`BlockAccount` shouldn't care about this kind of rules.

Reply:
@kfangw Hmm, I want to make restricting of `BaseReserve` to be more strictly, but I will accept your opinion.

Reply:
@spikeekips : It's not about being strict, but about separation of concerns (SoC).
`Save`'s responsibility is to write the data out. It should only check for things that prevents it from doing its job, like read-only fs, `nil` key, etc...

Reply:
@Geod24 What you mean 'Save's responsibility? If we have something to check in `BlockAccount`, we can do it in `Save()`.

Reply:
@spikeekips : No, if it gets to `Save` that means `BlockAccount` is already is an invalid state.
By responsibility I mean "role", "duty", etc... Or, in other words, what is the scope of this function ? Why don't we put all the consensus rules in `Save`, for example ?

Reply:
@Geod24 Whether putting the some checking routine in `Save()` or not, it's just matter of choice, what is the explicit rule can be exist? What is the 'role' or 'duty'? Who made these things  for our `Save()`?

Reply:
@spikeekips : One simple rule is about side effect. By isolating side effects as much as possible, code is more robust and testable. And `Save` function is a nest of side effects (two write to the storage and triggers the observer).